cpp/inc/bond/core/detail/mpl.h

```diff
@@ -17,11 +17,18 @@
 #include <utility>
 
 
-namespace bond
+namespace bond { namespace detail { namespace mpl
 {
 
-namespace detail { namespace mpl
+template <typename...> struct
+make_void
 {
+    using type = void;
+};
+
+template <typename... T>
+using void_t = typename make_void<T...>::type;
```

Comment:
Is `make_void` there to make `void_t` dependent on `T`?

Is the compiler allowed to "inline" `make_void`? If so, how do we prevent it from seeing that `make_void` doesn't depend on `T` and thus neither does `void_t`?

Reply:
Following the [`std::void_t`](http://en.cppreference.com/w/cpp/types/void_t)

> Until CWG 1558 (a C++14 defect), unused parameters in alias templates were not guaranteed to ensure SFINAE and could be ignored, so earlier compilers require a more complex definition of void_t, such as
> ```cpp
> template<typename... Ts> struct make_void { typedef void type;};
> template<typename... Ts> using void_t = typename make_void<Ts...>::type;
> ```